src/bundles.ts

```diff
@@ -112,15 +118,17 @@ export async function uploadRemoteBundle({
     return false;
   };
 
-  const tasks = [];
+  const tasks: FileInfo[][] = [];
   for (const bucketFiles of composeFilePayloads(options.files, maxPayload)) {
-    tasks.push(uploadFileChunks(bucketFiles));
+    tasks.push(bucketFiles);
   }
-
-  if (tasks.length) {
-    return (await Promise.all(tasks)).some(r => !!r);
-  }
-  return true;
+  const results = await pMap(
+    tasks,
+    async (task: FileInfo[]) => await uploadFileChunks(task),
+    { concurrency: UPLOAD_CONCURRENCY }
+  );
+  // Returning false if at least one result is false
+  return !results.some(r => !r);
```

Comment:
`nit`: it's kind of my own preference but in order to avoid double negation you can use:
```suggestion
  return results.every(r => !!r);
```

Reply:
Like the idea!